moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useCsvExport.js

```diff
@@ -51,13 +94,25 @@ const downloadFile = (fileContents) => {
  * Builds a record entry given a specific configuration and filters
  * @param {object} record - The record to build
  * @param {object} exportConfig - The export configuration
+ * @param {string} columnDownloadOption - The column download option selected by user, either "all" or "visible"
+ * @param {array} visibleColumns - The columns that are currently visible in the table
  * @return {object}
  */
-const buildRecordEntry = (record, exportConfig) => {
+const buildRecordEntry = (
+  record,
+  exportConfig,
+  columnDownloadOption,
+  visibleColumns
+) => {
   // Allocate an empty object
   const entry = {};
+  const columnsToExport =
+    // Based on user selection set columns to download as all or just currently visible
+    columnDownloadOption === "visible"
+      ? visibleColumns
+      : Object.keys(exportConfig);
```

Comment:
**For a future spinoff and out of scope**: This caught my eye since it seems to be the second time that `visibleColumns` is used to determine whether to include all or just the visible columns even though the project records coming into this function already have all columns or just the visible columns. 

I think that we could get away with iterating the fields in the record to get the needed `exportConfig` instead of iterating the columns in `columnsToExport`. I might be missing a reason that wouldn't work, though. 🤔